Add previous data on transform (#7102)
marmelab/react-admin: docs/CreateEdit.md

# The Create and Edit Views

> `<Resource>` maps URLs to components - it takes care of *routing*. When you set a component as the `create` prop for a Resource, react-admin renders that component when users go to the `/[resource]/create` URL. When you set a component as the `edit` prop for a resource, react-admin renders that component when users go to the `/[resource]/:id` URL.

Reviewer: There are a lot of unrelated and unnecessary changes here due to your markdown formatter probably. Can you please revert them?

**Tip**: The message will be translated.

> **Tip**: `<Edit>`'s transform prop can recieve below variables from second argument.
> 
> |  field         | Description                                                                                |
> | -------------- | ------------------------------------------------------------------------------------------ |
> | `previousData` | Previous data fetched or last updated on `dataProvider.update`                             |
> 

Reviewer: ```suggestion
**Tip**: `<Edit>`'s transform prop function also get the `previousData` in its second argument:
```

**Tip**: The message will be translated.

**Tip**: `<Edit>`'s transform prop function also get the `previousData` in its second argument:

```jsx
export const UserEdit = (props) => {

>     const transform = (data, { previousData }) => ({

Reviewer 2: I'd prefer that the transform function signature be `(data, previousData)`

Author: @fzaninotto 

Sorry, but I'd prefer my way, because it rarely  breaks user code when we need to add more optionalData compared to push it to the tail of arguments though the change may be seldom.
Increasing argument have risks when we want to add new argument, intermediate arguments can't change order without breaking change and even if they are not used in some cases, we may need to assign undefined when we call  or when we set no used variables' specification (like, (data, _, someEditControllerParameter) as event hander.

Argument addition as option is better way for me if the argument is really optional value. In this case, users can ignore previousData so, it's optional value.
But, I can't say that previousData must be optional form because I haven't found that we need more additional parameter in the future. My point is opposite to YAGNI, and  previousData may be just enough parameter in all edit controller context parameters to satisfy user demands. So, I'll fix that  if you still want to do nonetheless you have read this comment.

}
```

> **Tip**: `<SaveButton>`'s transform prop inside `<Edit>` can recieve below variables from second argument.
> 
> |  field         | Description                                                                                |
> | -------------- | ------------------------------------------------------------------------------------------ |
> | `previousData` | Previous data fetched or last updated on `dataProvider.update`                             |

Reviewer: ```suggestion
**Tip**: `<Edit>`'s transform prop function also get the `previousData` in its second argument:
```
